src/Security/Member.php

```diff
@@ -1702,4 +1710,46 @@ public function getHtmlEditorConfigForCMS()
         // If can't find a suitable editor, just default to cms
         return $currentName ? $currentName : 'cms';
     }
+
+    /**
+     * This needs to be public because it's referenced in getMemberPasswordField() and called in ConfirmedPasswordField
+     */
+    public static function generateRandomPassword(array $args = []): string
+    {
+        $password = '';
+        $validator = self::password_validator();
+        $minLength = $validator ? $validator->getMinLength() : 20;
+        $maxLength = $args['maxLength'] ?? 0;
+        if ($maxLength === 0) {
+            $maxLength = max($minLength, 20);
+        } else {
+            if ($maxLength < $minLength) {
+                throw new InvalidArgumentException('maxLength argument is less than PasswordValidator minLength');
+            }
+        }
+        $dummyMember = new Member();
+        // These are similar character sets that used in the default PasswordValidator
+        $lc = 'abcdefghijklmnopqrstuvwyxz';
+        $uc = strtoupper($lc);
+        $num = '0123456789';
+        $pun = '!@#$%^&*()_+-=[]{};:,./<>?';
+        $chars = $lc . $uc . $num . $pun;
```

Comment:
Why not simply do this? We're not using the intermediate variables anywhere.
```suggestion
        $chars = 'abcdefghijklmnopqrstuvwyxzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()_+-=[]{};:,./<>?';
```

Reply:
Update now that I've looked at how password validator works:

It looks like we _should_ explicitly be using those intermediate variables, and explicitly adding some from each group, to _ensure_ we pass validation instead of just leaving it up to chance.

Reply:
Updated